### ISSUE TITLE
New error, despite not changing anything: "Your Private DNS setting is set. This may cause connectivity issues."

### ISSUE THREAD

**Issue by e-t-l** [blokadaorg/blokada#948]
Since Blokada 5 (F-Droid) updated to v22.3.8 three days ago, I've received the following popup error every time I open Blokada: "Ooops! Your Private DNS setting is set. This may cause connectivity issues. Please turn it off in Settings."

When I open the Private DNS android Settings page, I see it's set to Automatic. This is not something I've changed recently. The only recent change is that Blokada updates. Did this latest version change something to be more sensitive to Private DNS? Blokada seems to be continuing to handle traffic as usual.

I know in #503 we determined that Blokada will not work in conjunction with Private DNS on, but the "Automatic" setting (which seems to use DoT "when possible") seems to work just fine, as it appears to kick in for apps that have been whitelisted from Blokada.

--
One related issue I'm encountering is, while I have Cloudflare set as the DNS in Blokada, and (back when I had tried to set Private DNS to On) I had set Cloudflare set as android's DNS,* nevertheless when Private DNS is set to Automatic then OpenDNS is used to resolve domains.** This is almost certainly outside the scope of Blokada to fix, but it would be nice if there was a way to specify the DNS used by apps that are whitelisted in Blokada.
(*if I'm not explaining this well, see screenshot below for my current Private DNS settings.
**I confirmed this using the free [tenta.com/test](www.tenta.com/test) DNS identifier service in Firefox Mobile, which is whitelisted in Blokada)

*TLDR: Can someone confirm whether Blokada v22.3.8 changed something regarding Private DNS that is causing this popup? And can you confirm that my understanding of the relationship between Blokada, Private DNS, and whitelisted apps is correct?*

![Screenshot_20220922-102953](https://user-images.githubusercontent.com/40775958/191777781-3245228a-18c2-473a-b2b1-2729d99b6a6b.png)

**Comment by MorpheusUK** [comment]
I've been getting exactly the same with f-droid v22.3.8 since the upgrade whether mine is set to off or automatic. I'm using a Samsung Galaxy S22

**Comment by AIGeneratedUsername** [comment]
I confirm the bug with:
- Android 7.0
- Blokada 5
- installed from F-droid

**Comment by AML44** [comment]
I confirm too installed from F-Droid i get this error but when i download the .apk from blokada.org i don't enconter this problem.
Have you try to install from the official website ?

**Comment by e-t-l** [comment]
> Have you try to install from the official website ?

No I haven't. Would it install as an update if it's the same version number, or would I have to uninstall blokada first?

@AML44 Interesting, and what are your Private DNS settings?

**Comment by AML44** [comment]
For me i uninstall first the blokada application that i download from F-droid ans then i install the blokada dowbloaded from the original website.
For the private DNS in my smartphone it's set in automatic.

**Comment by e-t-l** [comment]
Out of curiosity, what happens if you set Private DNS to On?

Sep 23, 2022 3:44:37 PM AML44 ***@***.***>:

**Comment by AML44** [comment]
Dont know because i let the app in automatic mode

**Comment by MorpheusUK** [comment]
The issue does seem to be the f-droid build, downloaded the APK from the site and it doesn't have this issue.

**Comment by nieXas** [comment]
Are we forced to uninstall f-droid version??

**Comment by kar** [comment]
Thanks for reporting this. Sadly we do not maintain the f-droid listing anymore and don't have plans to do so. Blokada is open source so anyone is free to fork it and adapt if necessary. We recommend you to download the latest version from our website https://blokada.org.

**Comment by licaon-kter** [comment]
fyi https://gitlab.com/fdroid/fdroiddata/-/commit/51df6ec4eed02834ffdc2fa260a9320955cb11f2

**Comment by e-t-l** [comment]
@kar I'm actually still encountering this issue with this, using Blokada downloaded from Github...
